hazelcast/src/main/java/com/hazelcast/flakeidgen/impl/FlakeIdGeneratorProxy.java

```diff
@@ -226,10 +227,18 @@ int getNodeId(long nanoTime) {
 
                 // we ignore possible double initialization
                 this.nodeId = nodeId;
+                nextNodeIdUpdate = nanoTime + NODE_ID_UPDATE_INTERVAL_NS;
```

Comment:
Your code is OK, so I'm approving. This also works, I find it easier. If you like it you can replace it:
```java
        long nextNodeIdUpdate = this.nextNodeIdUpdate;
        int nodeId = this.nodeId;
        if (nextNodeIdUpdate <= nanoTime) {
            if (nodeId == NODE_ID_OUT_OF_RANGE) {
                return nodeId;
            }
            int newNodeId = getNodeEngine().getClusterService().getMemberListJoinVersion();
            assert newNodeId >= 0 : "newNodeId=" + newNodeId;
            newNodeId += nodeIdOffset;
            if (newNodeId != nodeId) {
                nodeId = newNodeId;

                // If our node ID is out of range, assign NODE_ID_OUT_OF_RANGE to nodeId
                if ((nodeId & -1 << bitsNodeId) != 0) {
                    outOfRangeMembers.add(getNodeEngine().getClusterService().getLocalMember().getUuid());
                    logger.severe("Node ID is out of range (" + nodeId + "), this member won't be able to generate IDs. "
                            + "Cluster restart is recommended.");
                    nodeId = NODE_ID_OUT_OF_RANGE;
                }

                // we ignore possible double initialization
                this.nodeId = nodeId;
                this.nextNodeIdUpdate = nanoTime + NODE_ID_UPDATE_INTERVAL_NS;
                if (logger.isFineEnabled()) {
                    logger.fine("Node ID assigned to '" + name + "': " + nodeId);
                }
            }
        }
```
It ensures that it's impossible to see no need to update before `nodeId` was assigned to some valid value.